packages/core/src/install/createPeerDependencyPatcher.ts

```diff
@@ -1,20 +1,27 @@
 import { PeerDependencyRules, ReadPackageHook } from '@pnpm/types'
+import matcher from '@pnpm/matcher'
 import isEmpty from 'ramda/src/isEmpty'
 
 export default function (
   peerDependencyRules: PeerDependencyRules
 ): ReadPackageHook {
-  const ignoreMissing = new Set(peerDependencyRules.ignoreMissing ?? [])
+  const ignoreMissingPatterns = [...new Set(peerDependencyRules.ignoreMissing ?? [])]
+  const ignoreVersionPatterns = [...new Set(peerDependencyRules.ignoreVersion ?? [])]
   return ((pkg) => {
     if (isEmpty(pkg.peerDependencies)) return pkg
     for (const [peerName, peerVersion] of Object.entries(pkg.peerDependencies ?? {})) {
-      if (ignoreMissing.has(peerName) && !pkg.peerDependenciesMeta?.[peerName]?.optional) {
+      if (
+        matcher(ignoreMissingPatterns)(peerName) &&
+        !pkg.peerDependenciesMeta?.[peerName]?.optional
+      ) {
         pkg.peerDependenciesMeta = pkg.peerDependenciesMeta ?? {}
         pkg.peerDependenciesMeta[peerName] = {
           optional: true,
         }
       }
-      if (
+      if (peerVersion !== '*' && matcher(ignoreVersionPatterns)(peerName)) {
+        pkg.peerDependencies![peerName] = '*'
+      } else if (
         peerDependencyRules.allowedVersions?.[peerName] &&
         peerVersion !== '*'
```

Comment:
both if statements check for `peerVersion !== '*'`. It can be moved to one place.